### DISCUSSION TITLE
How do we configure High Availability for Jan Server

### DISCUSSION THREAD

**Discussion by dhanan-np** [JanssenProject/jans#5014] [Q&A] [answered]
Hello There,

We want to setup our Jans server on 2 or more servers for HA. I need some help/guidance, how we can proceed ahead with this.
We will have a separate DB-Cluster of MySql. We will be using HAProxy as Load-Balancer. 
My Queries are primarily related to configuration file setup, for example-

1. Do we need to install any certificate on each of the instances or we can deploy only one at HAProxy. If we don't need certificate on all the servers, how do we disable the certificate installation process. And if needed to install on all of the instance server, how do we deploy the certificates which we have obtained from Verified CA(_no self-signed certificates_)
2. How do we skip the Apache installation on each of the server. 
3. From where I should download the installer for such cases.

**Comment by ossdhaval** [comment]
Hi @dhanan-np 

Thank you for starting this discussion. I wanted to check two things in this regard:

1. Is cloud-native deployment an option that you can explore?
2. If not then have you gone through [this document](https://docs.jans.io/v1.0.13/admin/planning/vm-cluster/)?

**Reply by dhanan-np** [reply]
Yes I have gone through that document and we are not doing it on Cloud-Native

**Reply by dhanan-np** [reply]
Some other questions -

- **how do we configure Redis Cache**? I didn't find any option in the installation process where it asks for any such things
- My installation will be based **on MySql Database**. Do I still need the **OpenDJ (with replication)**?

**Comment by moabu** [comment, marked as answer]
- With MySQL you shouldn't need Redis Cache. If you do later on you may add it using the TUI.
- No you do not need OpenDJ if you are using MySQL

We do not recommend the HA setup with VMs because we have found the trend that people fall back in keeping the setup updated with latest versions because the method of upgrading VMs requires some work and patching. We recommend you follow the CN setup for HA as you will get automatic upgrades, patches with a single `helm upgrade` command. 


With that said the HA VM setup might be possible but it is not tested:

The arch would have a central MySQL which two or more Jans VMs would connect to. On the first jans VM you would go through a normal [installation](https://docs.jans.io/stable/admin/install/vm-install/) which would connect to the MySQL and setup your persistence along the way. On the second and subsequent VMs you would run a snapshot of the first VM. 


There is a poll here https://github.com/JanssenProject/jans/discussions/5027 which you can vote on that would add support for VM HA setup natively.
